src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/BlockSyntaxExtensions.cs

```diff
@@ -62,6 +62,17 @@ internal static class BlockSyntaxExtensions
                 block.TryConvertToExpressionBody(languageVersion, preference, out var expression, out semicolonToken))
             {
                 arrowExpression = SyntaxFactory.ArrowExpressionClause(expression);
+
+                if (block.GetRequiredParent().Kind() == SyntaxKind.GetAccessorDeclaration)
+                {
+                    var comments = block.GetRequiredParent().GetLeadingTrivia().Where(t => t.IsSingleOrMultiLineComment());
+                    if (!comments.IsEmpty())
```

Comment:
would things go haywire if we just added the leading trivia unconditionally?  I would prefer we do that if we can because there are other forms of trivia (for example preprocessor directives `#if true`

Reply:
Yeah, a bunch of tests were failing because it added an extra line. Maybe I can change it so it checks that it isn't a white space or a line break instead of checking if it's a comment.

Reply:
That sounds reasonable - I think we have a helper for that - https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/ISyntaxFactsExtensions.cs,391